quota/BaseQuotaManager.js

```diff
@@ -0,0 +1,88 @@
+      logger.debug('Org whitelisted, Quota check skipped');
+      return CONST.QUOTA_API_RESPONSE_CODES.VALID_QUOTA;
+    } else {
+      logger.debug(`Platform is ${this.platform}`);
+      logger.debug(`Org/Subaccount id is ${orgOrSubaccountId}`);
+      logger.debug(`Plan id is ${planId}`);
+      let planName = _.find(catalog.plans, ['id', planId]).name;
+      let serviceName = _.find(catalog.plans, ['id', planId]).service.name;
+      let skipQuotaCheck = _.find(catalog.plans, ['id', planId]).metadata ? _.find(catalog.plans, ['id', planId]).metadata.skip_quota_check : undefined;
+      logger.debug(`Plan Name is ${planName}`);
+      logger.debug(`Service Name is ${serviceName}`);
+      logger.debug(`Skip Quota check: ${skipQuotaCheck}`);
+      if (skipQuotaCheck && skipQuotaCheck === true) {
+        return CONST.QUOTA_API_RESPONSE_CODES.VALID_QUOTA;
+      } else {
+        const quota = await this.quotaAPIClient.getQuota(orgOrSubaccountId, serviceName, planName, useSubaccountForQuotaCheck);
+        // Special cases:
+        // When obtained quota = 0, send message to customer – Not entitled to create service instance
+        // When obtained quota = -1, assume that the org is whitelisted and hence allow the creation
+        if (quota === 0) {
+          return CONST.QUOTA_API_RESPONSE_CODES.NOT_ENTITLED;
+        } else if (quota === -1) {
+          return CONST.QUOTA_API_RESPONSE_CODES.VALID_QUOTA;
+        } else {
+          const planIdsWithSameSKU = this.getAllPlanIdsWithSameSKU(planName, serviceName, catalog);
+          const instanceCount = await this.getInstanceCountonPlatform(orgOrSubaccountId, planIdsWithSameSKU);
+          logger.debug(`Number of instances are ${instanceCount} & Quota number for current org space and service_plan is ${quota}`);
+          return instanceCount >= quota ? CONST.QUOTA_API_RESPONSE_CODES.INVALID_QUOTA : CONST.QUOTA_API_RESPONSE_CODES.VALID_QUOTA;
+        }
+      }
+    }
+  }
+  async getInstanceCountonPlatform(orgOrSubaccountId, planGuids) {
+    throw new NotImplementedBySubclass(`getInstanceCountonPlatform - ${orgOrSubaccountId}, ${planGuids}`);
+  }
+  getAllPlanIdsWithSameSKU(planName, serviceName, serviceCatalog) {
+    const planManagerName = _.find(catalog.plans, ['name', planName]).manager.name;
+    const skuName = this.getSkuNameForPlan(planName);
+
+    logger.debug(`SKUName is ${skuName}`);
+    const planIdsWithSameSKU = [];
+    const service = _.find(serviceCatalog.services, ['name', serviceName]);
+    _.each(service.plans, plan => {
+      if (plan.name.endsWith(skuName) && plan.manager.name === planManagerName) {
+        planIdsWithSameSKU.push(plan.id);
+        logger.debug(`Found a plan with name as ${plan.name} which contains the skuName ${skuName}`);
+      }
+    });
+    logger.debug('sameskuplanids are ', planIdsWithSameSKU);
+    return planIdsWithSameSKU;
+  }
+
+  async isOrgWhitelisted(orgId) {
+    return false;
+  }
+
```

Comment:
Will it make sense to move this function to `K8SPlatformQuotaManager`, because IIUC, this is meant to handle the k8s consumption case and if we decide to handle some whitelisting for k8s case, this function will come handy. Also I think we can loose `async` here.

Reply:
Removing from Base class means child classes will have to necessarily need to implement this function, which is not intended, async I have put just to maintain parity.